app/js/benefitRisk/schemaService.js

```diff
@@ -588,11 +597,80 @@ define(['lodash', 'angular', 'ajv'], function (_, angular, Ajv) {
       };
     }
 
+    function extractPvfs(criteria) {
+      return _(criteria)
+        .mapValues((criterion) => _.omit(criterion.dataSources[0].pvf, 'range'))
+        .pickBy(isUsablePvf)
+        .value();
+    }
+
+    function isUsablePvf(pvf) {
+      return pvf.direction;
+    }
+
+    function extractRanges(criteria) {
+      return _(criteria)
+        .flatMap('dataSources')
+        .filter('pvf')
+        .keyBy('id')
+        .mapValues((dataSource) => dataSource.pvf.range)
+        .value();
+    }
+
+    function mergePvfs(scenario, pvfs) {
+      const updatedCriteria = updateCriteriaWithPvfs(
+        scenario.state.problem.criteria,
+        pvfs
+      );
+      return _.merge({}, scenario, {
+        state: {problem: {criteria: updatedCriteria}}
+      });
+    }
+
+    function updateCriteriaWithPvfs(scenarioCriteria, pvfs) {
+      return _.mapValues(scenarioCriteria, (scenarioCriterion, criterionId) => {
+        return scenarioCriterion.dataSources
```

Comment:
Didn't we run into a case where we needed to check more thoroughly if the criterion had a pvf?